File: CoreTransformer/TransitionAttributeValue.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

''' <summary>
''' Transition Attribute Value class
''' </summary>
''' <remarks></remarks>
Class TransitionAttributeValue
    Inherits AttributeValueBase

    Private m_TransitionGroupId As Integer

    Public Sub New(
        ByVal transitionAttributeTypeId As Integer,
        ByVal stratumId As Nullable(Of Integer),
        ByVal secondaryStratumId As Nullable(Of Integer),
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal transitionGroupId As Integer,
        ByVal stateClassId As Nullable(Of Integer),
        ByVal minimumAge As Nullable(Of Integer),
        ByVal maximumAge As Nullable(Of Integer),
        ByVal value As Double)

        MyBase.New(
            transitionAttributeTypeId,
            stratumId,
            secondaryStratumId,
            iteration,
            timestep,
            stateClassId,
            minimumAge,
            maximumAge,
            value)

        Me.m_TransitionGroupId = transitionGroupId

    End Sub

    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

End Class

File: src/Runtime/TransitionAttributeValue.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2021 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using SyncroSim.StochasticTime;

namespace SyncroSim.STSim
{
    internal class TransitionAttributeValue : STSimDistributionBase
    {
        private int m_TransitionAttributeTypeId;
        private int m_TransitionGroupId;
        private int? m_StateClassId;
        private int m_MinimumAge;
        private int m_MaximumAge;
        private int? m_TSTGroupId;
        private int? m_TSTMin;
        private int? m_TSTMax;

        public TransitionAttributeValue(
            int transitionAttributeTypeId, 
            int? stratumId, 
            int? secondaryStratumId, 
            int? tertiaryStratumId, 
            int? iteration, 
            int? timestep, 
            int transitionGroupId, 
            int? stateClassId, 
            int minimumAge, 
            int maximumAge,
            int? tstGroupId,
            int? tstMin,
            int? tstMax,
            double? value,
            int? distributionTypeId,
            DistributionFrequency? distributionFrequency,
            double? distributionSD,
            double? distributionMin,
            double? distributionMax) : base(
                iteration, timestep, stratumId, secondaryStratumId, tertiaryStratumId,
                value,
                distributionTypeId, distributionFrequency, distributionSD, distributionMin, distributionMax)
        {
            this.m_TransitionAttributeTypeId = transitionAttributeTypeId;
            this.m_TransitionGroupId = transitionGroupId;
            this.m_StateClassId = stateClassId;
            this.m_MinimumAge = minimumAge;
            this.m_MaximumAge = maximumAge;
            this.m_TSTGroupId = tstGroupId;
            this.m_TSTMin = tstMin;
            this.m_TSTMax = tstMax;
        }

        public int TransitionAttributeTypeId
        {
            get
            {
                return this.m_TransitionAttributeTypeId;
            }
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public int? StateClassId
        {
            get
            {
                return this.m_StateClassId;
            }
        }

        public int MinimumAge
        {
            get
            {
                return this.m_MinimumAge;
            }
        }

        public int MaximumAge
        {
            get
            {
                return this.m_MaximumAge;
            }
        }

        public int? TSTGroupId
        {
            get
            {
                return this.m_TSTGroupId;
            }
        }

        public int? TSTMin
        {
            get
            {
                return this.m_TSTMin;
            }
        }

        public int? TSTMax
        {
            get
            {
                return this.m_TSTMax;
            }
        }

        public override STSimDistributionBase Clone()
        {
            return new TransitionAttributeValue(
                this.TransitionAttributeTypeId,
                this.StratumId,
                this.SecondaryStratumId,
                this.TertiaryStratumId,
                this.Iteration,
                this.Timestep,
                this.TransitionGroupId,
                this.StateClassId,
                this.MinimumAge,
                this.MaximumAge,
                this.TSTGroupId,
                this.TSTMin,
                this.TSTMax,
                this.DistributionValue,
                this.DistributionTypeId,
                this.DistributionFrequency,
                this.DistributionSD,
                this.DistributionMin,
                this.DistributionMax);
        }
    }
}
